fe/fe-core/src/main/java/com/starrocks/external/hive/HiveMetaCache.java

```diff
@@ -468,20 +475,22 @@ public void refreshTable(HiveMetaStoreTableInfo hmsTable)
 
     public void refreshPartition(HiveMetaStoreTableInfo hmsTable, List<String> partNames) throws DdlException {
         GlobalStateMgr.getCurrentState().getMetastoreEventsProcessor().getEventProcessorLock().writeLock().lock();
-        try {
-            for (String partName : partNames) {
-                List<String> partValues = client.partitionNameToVals(partName);
-                HivePartitionKey key = new HivePartitionKey(hmsTable.getDb(), hmsTable.getTable(),
-                        hmsTable.getTableType(), partValues);
-                partitionsCache.put(key, loadPartition(key));
-                partitionStatsCache.put(key, loadPartitionStats(key));
-            }
-        } catch (Exception e) {
-            LOG.warn("refresh partition cache failed", e);
-            throw new DdlException("refresh partition cached failed: " + e.getMessage());
-        } finally {
-            GlobalStateMgr.getCurrentState().getMetastoreEventsProcessor().getEventProcessorLock().writeLock().unlock();
+        for (String partName : partNames) {
+            partitionRefreshExecutor.submit(() -> {
```

Comment:
I think this 'submit()' will not wait for the task to finish before exiting the method, you need to block each task until all executions are complete

Reply:
Done.